### PR TITLE
Add support for x264 and h264

### PR DESCRIPTION
**Pull request by duramato**

### CODE REVIEW

**Comment by SiCKRAGEB0T** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Give a description on what the PR is for.
- [x] Make sure you send a PR against the DEVELOP branch
- [x] Don't send big changes in one go. Split up big PRs into multiple smaller (easier manageble) PRs

Please make sure you also read [contribution guide](https://github.com/SiCKRAGETV/SickRage/blob/master/contributing.md) and followed all the steps.

Thanks!

**Comment by miigotu** [comment]
Cant just do that at random lol, look at RAWHDTV. All 108p will get recognized as RAWHDTV if I accepted this.

Most of these are resctrictive for a reason.

**Comment by duramato** [comment]
mmm ,and what about the other qualities?

**Comment by miigotu** [comment]
Whats the point? The only reason to EVER change these is if something is not getting detected properly, and it can be demonstrated on a large scale.

File: sickbeard/common.py [hunk 1]
```diff
@@ -270,22 +270,22 @@ def sceneQuality(name, anime=False):
             ret = Quality.SDTV
         elif checkName([r"web.?dl|web(rip|mux)", r"xvid|x26[45]|h.?26[45]"], all) and not checkName([r"(720|1080)[pi]"], all):
             ret = Quality.SDTV
-        elif checkName([r"(dvd(rip|mux)|b[rd](rip|mux)|blue?-?ray)(.ws)?.(xvid|divx|x26[45])"], any) and not checkName([r"(720|1080)[pi]"], all):
+        elif checkName([r"(dvd(rip|mux)|b[rd](rip|mux)|blue?-?ray)(.ws)?.(xvid|divx|[xh]26[45])"], any) and not checkName([r"(720|1080)[pi]"], all):
```
Comment (miigotu) [inline sickbeard/common.py]: `[xh].?26[45]`

File: sickbeard/common.py [hunk 2]
```diff
@@ -270,22 +270,22 @@ def sceneQuality(name, anime=False):
...
             ret = Quality.SDDVD
-        elif checkName([r"720p", r"hd.?tv", r"x26[45]"], all) or checkName([r"hr.ws.pdtv.x26[45]"], any) and not checkName(
+        elif checkName([r"720p", r"hd.?tv", r"[xh]26[45]"], all) or checkName([r"hr.ws.pdtv.[xh]26[45]"], any) and not checkName(
```
Comment (miigotu) [inline sickbeard/common.py]: `[xh].?26[45]`

File: sickbeard/common.py [hunk 3]
```diff
@@ -270,22 +270,22 @@ def sceneQuality(name, anime=False):
...
                 [r"1080[pi]"], all):
             ret = Quality.HDTV
-        elif checkName([r"720p|1080i", r"hd.?tv", r"mpeg-?2"], all) or checkName([r"1080[pi].hdtv", r"h.?26[45]"], all):
+        elif checkName([r"720p|1080i", r"hd.?tv", r"mpeg-?2"], all) or checkName([r"1080[pi].hdtv", r"[xh].?26[45]"], all):
```
Comment (miigotu) [inline sickbeard/common.py]: `h.26[45]`

File: sickbeard/common.py [hunk 4]
```diff
@@ -270,22 +270,22 @@ def sceneQuality(name, anime=False):
...
             ret = Quality.RAWHDTV
-        elif checkName([r"1080p", r"hd.?tv", r"x26[45]"], all):
+        elif checkName([r"1080p", r"hd.?tv", r"[xh]26[45]"], all):
```
Comment (miigotu) [inline sickbeard/common.py]: `[xh].?26[45]`

File: sickbeard/common.py [hunk 5]
```diff
@@ -270,22 +270,22 @@ def sceneQuality(name, anime=False):
...
             ret = Quality.FULLHDTV
-        elif checkName([r"720p", r"web.?dl|web(rip|mux)"], all) or checkName([r"720p", r"itunes", r"h.?26[45]"], all):
+        elif checkName([r"720p", r"web.?dl|web(rip|mux)"], all) or checkName([r"720p", r"itunes", r"[xh].?26[45]"], all):
             ret = Quality.HDWEBDL
-        elif checkName([r"1080p", r"web.?dl|web(rip|mux)"], all) or checkName([r"1080p", r"itunes", r"h.?26[45]"], all):
+        elif checkName([r"1080p", r"web.?dl|web(rip|mux)"], all) or checkName([r"1080p", r"itunes", r"[xh].?26[45]"], all):
             ret = Quality.FULLHDWEBDL
-        elif checkName([r"720p", r"blue?-?ray|hddvd|b[rd](rip|mux)", r"x26[45]"], all):
+        elif checkName([r"720p", r"blue?-?ray|hddvd|b[rd](rip|mux)", r"[xh]26[45]"], all):
             ret = Quality.HDBLURAY
```
Comment (miigotu) [inline sickbeard/common.py]: `[xh].?26[45]`

File: sickbeard/common.py [hunk 6]
```diff
@@ -270,22 +270,22 @@ def sceneQuality(name, anime=False):
...
-        elif checkName([r"1080p", r"blue?-?ray|hddvd|b[rd](rip|mux)", r"x26[45]"], all):
+        elif checkName([r"1080p", r"blue?-?ray|hddvd|b[rd](rip|mux)", r"[xh]26[45]"], all):
             ret = Quality.FULLHDBLURAY
```
Comment (miigotu) [inline sickbeard/common.py]: `[xh].?26[45]`

**Comment by WebSpider** [comment]
Travis failed because with the regex changes, we don't meet our own tests anymore. This means we don't detect releases that we think we should :)

**Comment by miigotu** [comment]
@WebSpider thats why I noted the changes that need made =P

File: sickbeard/common.py [hunk 7]
```diff
@@ -270,22 +270,22 @@ def sceneQuality(name, anime=False):
...
             ret = Quality.SDTV
-        elif checkName([r"(dvd(rip|mux)|b[rd](rip|mux)|blue?-?ray)(.ws)?.(xvid|divx|x26[45])"], any) and not checkName([r"(720|1080)[pi]"], all):
+        elif checkName([r"(dvd(rip|mux)|b[rd](rip|mux)|blue?-?ray)(.ws)?.(xvid|divx|[xh].?26[45])"], any) and not checkName([r"(720|1080)[pi]"], all):
             ret = Quality.SDDVD
-        elif checkName([r"720p", r"hd.?tv", r"x26[45]"], all) or checkName([r"hr.ws.pdtv.x26[45]"], any) and not checkName(
+        elif checkName([r"720p", r"hd.?tv", r"[xh].?26[45]"], all) or checkName([r"hr.ws.pdtv.[xh]26[45]"], any) and not checkName(
                 [r"1080[pi]"], all):
             ret = Quality.HDTV
-        elif checkName([r"720p|1080i", r"hd.?tv", r"mpeg-?2"], all) or checkName([r"1080[pi].hdtv", r"h.?26[45]"], all):
+        elif checkName([r"720p|1080i", r"hd.?tv", r"mpeg-?2"], all) or checkName([r"1080[pi].hdtv", r"h.26[45]"], all):
```
Comment (miigotu) [inline sickbeard/common.py]: why removed the ? here?

File: sickbeard/common.py [hunk 8]
```diff
@@ -270,22 +270,22 @@ def sceneQuality(name, anime=False):
             ret = Quality.SDTV
         elif checkName([r"web.?dl|web(rip|mux)", r"xvid|x26[45]|h.?26[45]"], all) and not checkName([r"(720|1080)[pi]"], all):
             ret = Quality.SDTV
-        elif checkName([r"(dvd(rip|mux)|b[rd](rip|mux)|blue?-?ray)(.ws)?.(xvid|divx|x26[45])"], any) and not checkName([r"(720|1080)[pi]"], all):
+        elif checkName([r"(dvd(rip|mux)|b[rd](rip|mux)|blue?-?ray)(.ws)?.(xvid|divx|[xh].?26[45])"], any) and not checkName([r"(720|1080)[pi]"], all):
             ret = Quality.SDDVD
-        elif checkName([r"720p", r"hd.?tv", r"x26[45]"], all) or checkName([r"hr.ws.pdtv.x26[45]"], any) and not checkName(
+        elif checkName([r"720p", r"hd.?tv", r"[xh].?26[45]"], all) or checkName([r"hr.ws.pdtv.[xh]26[45]"], any) and not checkName(
```
Comment (miigotu) [inline sickbeard/common.py]: and why skipped the `.?` here?

File: sickbeard/common.py (hunk 7 above)
Comment (duramato) [inline sickbeard/common.py]: you told me to do so :P

File: sickbeard/common.py (hunk 8 above)
Comment (duramato) [inline sickbeard/common.py]: ?

Comment (miigotu) [inline sickbeard/common.py]: the second check in this line:
`r"hr.ws.pdtv.[xh]26[45]"`
to
`r"hr.ws.pdtv.[xh].?26[45]"`

**Pull request merged**